crates/core/tedge_agent/src/restart_manager/actor.rs

```diff
@@ -206,17 +208,38 @@ impl RestartManagerActor {
 
     async fn get_restart_operation_commands(&self) -> Result<Vec<Command>, RestartManagerError> {
         let mut vec = vec![];
-        // sync first
-        let mut sync_command = Command::new(SUDO);
-        sync_command.arg(SYNC);
-        vec.push(sync_command);
 
         // reading `config_dir` to get the restart command or defaulting to `["init", "6"]'
         let system_config = SystemConfig::try_new(&self.config.config_dir)?;
 
-        let mut command = Command::new(SUDO);
-        command.args(system_config.system.reboot);
-        vec.push(command);
+        // Check if sudo command is available
+        match Command::new(SUDO)
+            .stdout(Stdio::null())
+            .stderr(Stdio::null())
+            .spawn()
+        {
+            Err(err) if err.kind() == ErrorKind::NotFound => {
+                let sync_command = Command::new(SYNC);
+                vec.push(sync_command);
+
+                let mut args = system_config.system.reboot.iter();
+
+                let mut command =
+                    Command::new(args.next().map(|s| s.to_string()).unwrap_or_default());
+                command.args(args);
+                vec.push(command);
+            }
+            _ => {
```

Comment:
This conditional arm includes the case when command `sudo` returns an error except NotFound. But the following code uses `sudo`, which sounds not correct to me.

I would remove `if err.kind() == ErrorKind::NotFound` and just not using `sudo` if any error occurs after execute `sudo`.